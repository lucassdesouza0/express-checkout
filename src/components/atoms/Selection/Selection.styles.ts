import styled from "@emotion/styled";

export const Section = styled("div")`
  display: flex;
  flex-flow: column wrap;
  padding: 10px;
`;

export const SelectionSection = styled("div")`
  display: flex;
  flex-flow: column wrap;
  flex-direction: column;
  align-items: center;
  overflow-y: auto;
  padding: 10px;
`;

export const Select = styled("select")`
  padding: 10px 20px;
  border-radius: 5px;
  background-color: transparent;
  font-size: 16px;
  border: none;
  box-shadow: 2px 2px 10px 5px #eee;
  margin: 10px 10px;

  :focus {
    outline: none;
  }
`;

export const Input = styled("input")`
  padding: 10px;
  border-radius: 5px;
  background-color: ${(props) => (props.disabled ? "#dddddd" : "transparent")};
  font-size: 16px;
  border: none;
  box-shadow: 2px 2px 10px 5px #eee;
  margin: 0 10px;
  max-width: 50px;
`;
