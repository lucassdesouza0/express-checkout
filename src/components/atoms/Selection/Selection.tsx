/* tslint:disable */
import React, { useContext, useEffect, useState } from "react";
import Link from "next/link";
import { useRouter } from "next/router";
import { useTranslation } from "next-i18next";
import { ContextType, ProductContext, OrderProps } from "pages/_app";

import Button from "atoms/Button/Button";
import { Section, SelectionSection, Select, Input } from "./Selection.styles";

interface SelectionProps {
  product?: OrderProps;
}

const Selection = ({ product }: SelectionProps) => {
  const [size, setSize] = useState<string>("");
  const [price, setPrice] = useState<number>();
  const [quantity, setQuantity] = useState<number>(1);
  const { productsContext, setProductsContext } = useContext(
    ProductContext
  ) as ContextType;
  const router = useRouter();
  const { t } = useTranslation("common");

  const variants = product?.variants;

  useEffect(() => {
    if (price && size)
      setProductsContext({
        product: {
          name: product?.name,
          variants: product?.variants,
          datail: { size: size, price: quantity * price, quantity: quantity },
        },
        products: productsContext.products,
      });
  }, [size, price, quantity]);

  function handleSizeChange(event: React.ChangeEvent<any>): void {
    let value;
    if (variants) {
      value = variants[event.target.value];
      setSize(value.size);
      setPrice(value?.price);
    }
  }

  function handleQuantityChange(event: React.ChangeEvent<any>): void {
    setQuantity(event.target.value);
  }

  return (
    <Section>
      <SelectionSection>
        <label>{t("packing-size")}</label>
        <Select name="ages" onChange={handleSizeChange} defaultValue={"sizes"}>
          <option value="sizes" disabled data-testid="select-option">
            sizes
          </option>
          {variants?.map((a, index) => (
            <option
              label={a.size}
              value={index}
              key={`${a}-${index}`}
              data-testid={`select-option-${a.size}`}
            >
              {a.size}
            </option>
          ))}
        </Select>
        <form
          name="quantity"
          onChange={handleQuantityChange}
          defaultValue={"sizes"}
        >
          <Input
            type="number"
            name="quantity"
            min="1"
            disabled={!price && !size}
          />
          <label htmlFor="quantity" data-testid="select-option">
            (unit)
          </label>
        </form>
        <Button
          type="button"
          onClick={() => router.push("checkout")}
          disabled={false}
        >
          <Link href="/checkout">{t("kart")}</Link>
        </Button>
      </SelectionSection>
    </Section>
  );
};

export default Selection;
