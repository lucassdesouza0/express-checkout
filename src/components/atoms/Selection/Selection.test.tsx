import React from "react";
import { screen, render, fireEvent } from "@testing-library/react";
import testRender from "../../../tests/render";

import mock from "../../../tests/__mocks__/product";

import Selection from "./Selection";

describe("Button", () => {
  const selectItem = mock.variants[0].size;
  test("it should mount", () => {
    testRender(<Selection product={mock} />);

    const select = screen.getAllByTestId("select-option");

    expect(select[0]).toBeInTheDocument();
  });

  test("it should match snapshot", () => {
    const { asFragment } = render(<Selection product={mock} />);
    expect(asFragment()).toMatchSnapshot();
  });
});
