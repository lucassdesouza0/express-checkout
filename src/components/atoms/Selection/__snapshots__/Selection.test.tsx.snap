// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button it should match snapshot 1`] = `
<DocumentFragment>
  <div
    class="css-1px0wn7-Section eui42cl3"
  >
    <div
      class="css-16xtfp9-SelectionSection eui42cl2"
    >
      <label>
        packing-size
      </label>
      <select
        class="css-12mkdmf-Select eui42cl1"
        name="ages"
      >
        <option
          data-testid="select-option"
          disabled=""
          selected=""
          value="sizes"
        >
          sizes
        </option>
        <option
          data-testid="select-option-3kg"
          label="3kg"
          value="0"
        >
          3kg
        </option>
        <option
          data-testid="select-option-10kg"
          label="10kg"
          value="1"
        >
          10kg
        </option>
        <option
          data-testid="select-option-20kg"
          label="20kg"
          value="2"
        >
          20kg
        </option>
      </select>
      <form
        name="quantity"
      >
        <input
          class="css-1xpb5t7-Input eui42cl0"
          disabled=""
          min="1"
          name="quantity"
          type="number"
        />
        <label
          data-testid="select-option"
          for="quantity"
        >
          (unit)
        </label>
      </form>
      <button
        class="css-hhtkyx-Button enhitpr0"
        name="button"
        type="button"
      >
        <a
          href="/checkout"
        >
          kart
        </a>
      </button>
    </div>
  </div>
</DocumentFragment>
`;
